RenderStatePackager/src/RenderStatePackager.cpp

```diff
@@ -25,12 +25,156 @@
  */
 
 #include "RenderStatePackager.hpp"
+#include "GraphicsAccessories.hpp"
+#include "BasicMath.hpp"
 #include "DefaultRawMemoryAllocator.hpp"
 #include "DynamicLinearAllocator.hpp"
+#include "SerializedPipelineState.h"
+#include "FileSystem.hpp"
+#include "FileWrapper.hpp"
 
 namespace Diligent
 {
 
+struct BytecodeDumper
+{
+    class WorkingDirectory
+    {
+    public:
+        WorkingDirectory(const Char* Path) :
+            m_CurrentPath{Path} {}
+
+
+        void Push(const Char* Path)
+        {
+            m_CurrentPath = m_CurrentPath + FileSystem::GetSlashSymbol() + Path;
```

Comment:
I highly recommend rework this with std::stack. This will make it marginally slower, but way more robust and error-prone